core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/composite/UnionMoveSelectorFactory.java

```diff
@@ -52,20 +48,6 @@ public MoveSelector<Solution_> buildBaseMoveSelector(HeuristicConfigPolicy<Solut
             }
             selectorProbabilityWeightFactory = ConfigUtils.newInstance(config,
                     "selectorProbabilityWeightFactoryClass", config.getSelectorProbabilityWeightFactoryClass());
-        } else if (randomSelection) {
```

Comment:
Removing this seems wierd

Reply:
because this is the code that calls innerMoveSelectorConfig.getFixedProbabilityWeight()